src/cmd/ksh93/tests/modifiers.sh

```diff
@@ -44,3 +44,41 @@ unset foo
 foo="non-null value"
 [[ ${foo:+bar} == "bar" ]]  || log_error  '${foo:+bar} not bar when foo is not null'
 [[ ${foo+bar} == "bar" ]]  || log_error  '${foo+bar} not bar when foo is not null'
+
+# Check for regressions on issue #475 where parens after `-', `+', and `=' were causing syntax errors
+unset foo
+[[ ${foo:-(bar)} == "(bar)" ]]  || log_error  '${foo:-(bar)} not (bar) when foo is not set'
+
+unset foo
+[[ ${foo-(bar)} == "(bar)" ]] || log_error '${foo-(bar)} not (bar) when foo is not set'
+
+unset foo
+[[ ${foo-(wor)d} == "(wor)d" ]] || log_error '${foo-(wor)d} not (wor)d when foo is not set'
+
+unset foo
+[[ ${foo-w(or)d} == "w(or)d" ]] || log_error '${foo-w(or)d} not w(or)d when foo is not set'
+
+unset foo
+[[ ${foo-w(ord} == "w(ord" ]] || log_error '${foo-w(ord} not w(ord when foo is not set'
+
+ unset foo
```

Comment:
Nitpick: Empty space at beginning of line.

Reply:
Good catch! Fixed.